ex18/main.go

```diff
@@ -1,5 +1,42 @@
 package main
 
-  func main() {
+import (
+	"fmt"
+	"sync"
+)
 
-  }
+type Counter struct {
+	count int
+	mu    sync.Mutex
+}
```

Comment:
It's ok but the easiest answer on this task would be using https://pkg.go.dev/sync/atomic